src/pybind/mgr/mgr_module.py

```diff
@@ -330,6 +330,70 @@ def device_class_counts(self) -> Dict[str, int]:
 HandlerFuncType = Callable[..., Tuple[int, str, str]]
 
 
+def _extract_target_func(
+    f: HandlerFuncType
+) -> Tuple[HandlerFuncType, Dict[str, Any]]:
+    """In order to interoperate with other decorated functions,
+    we need to find the original function which will provide
+    the main set of arguments. While we descend through the
+    stack of wrapped functions, gather additional arguments
+    the decorators may want to provide.
+    """
+    wrapped = getattr(f, "__wrapped__", None)
```

Comment:
just in case but __wrapped__ only exists if the the method was wrapped by using the @wrap decorator